### PR TITLE
Various improvements to PoC to increase the chances of getting receipts/witnesses

### PR DESCRIPTION
**Pull request by Vagabond**
See commit messages for details.

### CODE REVIEW

File: src/poc/miner_onion_server.erl [hunk 1]
```diff
@@ -326,22 +318,53 @@ wait_until_next_block() ->
             ok
     end.
 
+wait_for_block(Fun, Count) when Count > 0 ->
+    ok = blockchain_event:add_handler(self()),
+    wait_for_block_(Fun, Count).
```
Comment (evanmcc) [inline src/poc/miner_onion_server.erl]: should we unregister after this completes?

Comment (evanmcc) [reply src/poc/miner_onion_server.erl]: it's only called in a spawn, is the event system going to unregister it automatically after it exits?

**Comment by Vagabond** [comment]
> this seems reasonable; we should schedule some sort of refactoring pass for the onion server, it's hard to follow.

Agreed.
